src/bootstrap/src/core/config/config.rs

```diff
@@ -1314,7 +1314,13 @@ impl Config {
 
         // Set flags.
         config.paths = std::mem::take(&mut flags.paths);
-        config.skip = flags.skip.into_iter().chain(flags.exclude).collect();
+        config.skip = flags
+            .skip
+            .into_iter()
+            .chain(flags.exclude)
+            .map(|p| t!(config.src.join(p).canonicalize()))
```

Comment:
This doesn't seem to work for me. I tried:

```
./x test ui --stage 1 --skip tests\ui\traits\fn-pointer\bare-fn-no-impl-fn-ptr-99875.rs
```

It still ran the test. I know because it fails on my machine.

Reply:
I didn't change anything on the test execution logic. Does skipping works on master?

Reply:
Possibly it's an issue with the test framework itself.

Reply:
> I didn't change anything on the test execution logic. Does skipping works on master?

Yes it works on master. But not with an absolute path, only with a subpath.

Reply:
> Yes it works on master. But not with an absolute path, only with a subpath.

That could be the reason, will check.

Reply:
Ah yeah, libtest uses test names like (on Windows):

```
[ui] tests\\ui\\traits\\fn-pointer\\bare-fn-no-impl-fn-ptr-99875.rs
```

It then calls the string `str::contains` on the test name for each `--skip` path https://github.com/rust-lang/rust/blob/4606a4d6fab93e53b54a4312c82e715eab8e122d/library/test/src/lib.rs#L491 So it'll only match relative paths and then only if the path separators match the platform's default.

So either bootstrap would need to make the path relative again or else this actually needs to be fixed in the test library.

Reply:
https://github.com/rust-lang/rust/pull/134209/commits/2421826cd3115aa37fbc3164420bfb8abe68cf60 should make everything work as expected.